fea-rs/src/compile/compiler.rs

```diff
@@ -23,27 +23,27 @@ const DEFAULT_N_MESSAGES_TO_PRINT: usize = 100;
 /// This is intended as the principal public API for this crate.
 ///
 /// ```no_run
-/// # use fea_rs::Compiler;
+/// # use fea_rs::{Compiler, compile::{NopFeatureProvider, NopVariationInfo}};
 /// # fn make_glyph_map() -> fea_rs::GlyphMap { todo!() }
 /// let glyph_map = make_glyph_map();
-/// let my_font_bytes = Compiler::new("path/to/features.fea", &glyph_map)
+/// let my_font_bytes = Compiler::<'_, NopFeatureProvider, NopVariationInfo>::new("path/to/features.fea", &glyph_map)
```

Comment:
I don't exactly know what the consequences of this changes are (or why is it important to not have the `dyn` and such), but merely looking at the API, this appears to make constructing a new fea-rs Compiler a lot more verbose..

Reply:
I agree that this feels like a worse API, I'll play around a bit and see if there's a middle ground.

Reply:
dyn is an extra layer of indirection to make calls. One of Rusts strengths is that you don't have to do that very often. Every execution of a compiler will do ever so slightly less work.

Ideally we'd have a default but iiuc from https://rust-lang.github.io/rfcs/0213-defaulted-type-params.html that is not currently possible.